micro-rdk/src/common/data_store.rs

```diff
@@ -164,36 +225,26 @@ impl DataStore for StaticMemoryDataStore {
 
         Ok(())
     }
-    fn read_next_message(
-        &mut self,
-        collector_key: &ResourceMethodKey,
-    ) -> Result<BytesMut, DataStoreError> {
-        let buffer_index = self.get_index_for_collector(collector_key)?;
-        let buffer = &self.buffers[buffer_index];
-        if buffer.is_empty() {
-            return Ok(BytesMut::with_capacity(0));
-        }
-
-        let mut cons = unsafe { Consumer::new(buffer) };
-        let (left, right) = cons.as_slices();
-        let mut chained = Buf::chain(left, right);
-        let encoded_len = decode_varint(&mut chained)? as usize;
-        let advance = length_delimiter_len(encoded_len);
-        unsafe { cons.advance(advance) };
-
-        let mut msg_bytes = BytesMut::with_capacity(encoded_len);
-        unsafe {
-            msg_bytes.set_len(encoded_len);
-        }
-        cons.pop_slice(&mut msg_bytes);
-        Ok(msg_bytes)
-    }
 
     fn from_resource_method_keys(
         collector_keys: Vec<ResourceMethodKey>,
     ) -> Result<Self, DataStoreError> {
         Self::new(collector_keys)
     }
+
+    fn get_reader(
```

Comment:
Is it legal for a Producer to put to the underlying `Rb` between two calls to `read_next_message` on the returned `StaticMemoryDataStore`? If so, I think there may be a concern that the position of `head` may be invalidated if old messages need to  be dropped.

Reply:
It is not legal, I'll put protections to stop this